MapboxNavigation/NSAttributedString.swift

```diff
@@ -1,12 +1,25 @@
 import Foundation
 
 extension NSAttributedString {
-    static func + (left: NSAttributedString, right: NSAttributedString) -> NSAttributedString
-    {
+    static func + (left: NSAttributedString, right: NSAttributedString) -> NSAttributedString {
         let result = NSMutableAttributedString()
         result.append(left)
         result.append(right)
         return result
     }
+}
 
+extension NSMutableAttributedString {
+    func canonicalizeAttachments() {
+        enumerateAttribute(.attachment, in: NSRange(location: 0, length: length), options: []) { (value, range, stop) in
+            guard let attachment = value as? NSTextAttachment, type(of: attachment) != NSTextAttachment.self else {
```

Comment:
@1ec5 this guard... it doesn't read right!!! Would this be why some of the shields are not showing up sometimes?

Reply:
@vincethecoder To me, it's reading as "The attachment is a subclass of `NSAttachment`, but not the actual class `NSAttachment.self`

Reply:
@JThramer Yes, I'm reading this as _we are looking for an attachment which is a subclass of `NSTextAttachment` but isn't of type `NSTextAttachment`_. But I suppose that's why we are attempt to cannoicalize the attachment(s).

Reply:
@vincethecoder yeah, this is likely because of limitations of the `NSAttributedString` logic in CarPlay.

Reply:
The application marshals the attributed string over to the CarPlay simulator or device for rendering. At least for the simulator if not also for the device, the attributed string is marshaled over XPC. It isn’t possible to marshal over instances of our custom NSTextAttachment subclasses, especially not the executable code in these subclasses. “Canonicalizing” the attributed string consists of replacing each subclassed attachment with a plain-vanilla NSTextAttachment that is sure to marshal over correctly.

There are other limitations to attributed strings in CarPlay maneuver instructions, such as being unable to change fonts or font sizes. However, those limitations aren’t limited to CarPlay. For example, on macOS, [you can use attributed strings in title bars](https://stackoverflow.com/a/29336473/4585461), but many attributes are ignored.